File: 0038-count-and-say/0038-count-and-say.cs
public class Solution {
    public string CountAndSay(int n) {
        string ans = "1";
        for(int i = 0; i < n - 1; i ++){
            ans = this.helper(ans);
        }
        
        return ans;
        
    }
    
    
    private string helper(string s){
        string final = "";
        
        int counter = 1;
        for(int i = 1; i < s.Length; i ++){
            if(s[i] == s[i-1]){
                counter++;
            }
            else{
                final +=  counter.ToString() + s[i-1].ToString();
                counter = 1;
            }
        }
        
        final += counter.ToString() + s[s.Length - 1].ToString();
        
        return final;
        
    }
}

File: 0038-count-and-say/0038-count-and-say.py
class Solution(object):
    def countAndSay(self, n):
        """
        :type n: int
        :rtype: str
        """
        
        def helper(a):
            final = ""
            counter = 1
            
            for i in range(1, len(a)):
                if a[i - 1] == a[i]:
                    counter += 1
                else:
                    final += str(counter) + a[i - 1] 
                    counter = 1
            
            final += str(counter) + a[-1]
            return final
        
        ans = "1"
        for i in range(1, n):
            ans = helper(ans)
        
        return ans